common/djangoapps/terrain/stubs/http.py

```diff
@@ -217,6 +218,8 @@ def _format_msg(self, format_str, *args):
         `format_str` is a string with old-style Python format escaping;
         `args` is an array of values to fill into the string.
         """
+        if not args:
+            format_str = urllib.unquote(format_str)
```

Comment:
@robrap We are using `format_str % args` on line 226, Let's say we have `%` sign in `format_str`, in this case course_id e.g. `'v1%3edX%2BDemoX%2BDemo_Course` includes that but as we know the `args` are empty which will make it fail by saying `TypeError: not enough arguments for format string`. Hope that helps to understand the problem ?